scripts_staging/Win_StorageCraftImageManager_Status.ps1

```diff
@@ -0,0 +1,983 @@
+#Set the HSRShareUser/HSRSharePass variables to match ShareUser/SharePass
+if ((Test-Path env:shareuser) -and (-not(Test-Path env:hsrshareuser))){$env:hsrshareuser = $env:shareuser}
+if ((Test-Path env:sharepass) -and (-not(Test-Path env:hsrsharepass))){$env:hsrsharepass = $env:sharepass}
+
+#Set Parameter Defaults
+if (!$SkipWarnFTP) {$WarnFTP = $True}
+if (!$SkipWarnHSR) {$WarnHSR = $True}
+if (!$SkipFileSystemChecks) {$FileSystemChecks = $True}
+if (!$SkipFreeSpaceChecks) {$FreeSpaceChecks = $True}
+if (!$SkipEventLogChecks) {$EventLogChecks = $True}
+if (!$FreeSpaceWarnThreshold) {$FreeSpaceWarnThreshold = 5}
+if (!$FreeSpaceAlertThreshold) {$FreeSpaceAlertThreshold = 10}
+
+###------------------------------Declare Functions----------------------------------###
+###---------------------------------------------------------------------------------###
+
+#Takes in an Index (from a hsr** or ftp** Table), Returns [String] containing the Target Path
+Function Get-TargetPath ([Int] $Index) { #Returns String
+
+    ForEach ($drTargetPath in $dtTargetPaths) {
+
+        if ($drTargetPath["Index"] -eq $Index) {
+            Return $drTargetPath["Path"]
+        }
+    }
+}
+
+#Takes in an Index (from a w*Files Table), Returns DataRow containing the Watch Path
+Function Get-WatchPath ([Int] $Index) { #Returns DataRow
+
+    ForEach ($drWatchPath in $dtWatchPaths) {
+
+        if ($drWatchPath["Index"] -eq $Index) {
+            Return $drWatchPath
+        }
+    }
+
+}
+
+#Takes in a w*Sets Table and Index, Returns DataRow containing the Watch Set
+Function Get-WatchSet ([System.Data.DataTable] $Table, [Int] $Index) { #Returns DataRow
+
+    ForEach ($drWatchSet in $Table) {
+
+        if ($drWatchSet["Index"] -eq $Index) {
+            Return $drWatchSet
+        }
+    }
+}
+
+#Takes in the Index from the WatchPaths Table and returns [Boolean] $True if a w*Files Table Exists, otherwise returns $False.
+Function Check-WatchPathTable ([Int] $Index) { #Returns Boolean
+
+    $Result = $False
+
+    ForEach ($drTableWP in $dtTableList){
+    
+        [Int]$CurrentTableIndexWP = $drTableWP["name"] -replace "[^0-9]",'' #Extract the index number from the table name
+        $CurrentTableNameWP = $drTableWP["name"]
+        if (($CurrentTableIndexWP -eq $Index) -and ($CurrentTableNameWP.Substring(0,1) -eq "w")) {
+
+            $Result = $True
+            Break
+        }
+    }
+    
+    Return $Result
+}
+###---------------------------------------------------------------------------------###
+
+#Initialize Variables
+$NowTime = [DateTime]::Now
+$AlertLevel = 0 #0 = Pass, 1 = Informational, 2 = Warning, 3 = Error
+$AlertText = ''
+
+####-------------------------------###
+#          Database Checks           #
+####-------------------------------###
+
+#Image Manager will have the DB open exclusively. Copy the DB to Temp.
+$IMDBPath = "C:\Windows\temp\Imagemanager.mdb"
```

Comment:
Any reason you're using hard coded windows\temp folder?

Can we use the recommended folders? https://docs.tacticalrmm.com/contributing_community_scripts/#good-habits